pgjdbc/src/main/java/com/yugabyte/ysql/ClusterAwareLoadBalancer.java

```diff
@@ -49,11 +51,14 @@ public static ClusterAwareLoadBalancer instance() {
   public ClusterAwareLoadBalancer() {
   }
 
-  public static ClusterAwareLoadBalancer getInstance() {
+  public static ClusterAwareLoadBalancer getInstance(int refreshListSeconds) {
     if (instance == null) {
       synchronized (ClusterAwareLoadBalancer.class) {
         if (instance == null) {
           instance = new ClusterAwareLoadBalancer();
+          instance.refreshListSeconds =
+              refreshListSeconds > 0 && refreshListSeconds <= LoadBalanceProperties.MAX_REFRESH_INTERVAL ?
```

Comment:
What is the idea behind MAX_REFRESH_INTERVAL capped to 10 mins. If the user provides more than that this code will ignore it and will use the DEFAULT_REFRESH_INTERVAL which is 5 minutes. Right? So just trying to understand that why do we need MAX and not allowing the user to specify a larger interval?

Reply:
The number 10 is just a number I came up with for max refresh interval in minutes. So if the app specifies higher value, the driver resets it back to the default which is 5 min. This is done to ensure we don't set a really high refresh interval which makes it ineffective.